python/ray/dataframe/dataframe.py

```diff
@@ -286,43 +288,96 @@ def groupby(self, by=None, axis=0, level=None, as_index=True, sort=True,
             axis: The axis to groupby.
             level: The level of the groupby.
             as_index: Whether or not to store result as index.
+            sort: Whether or not to sort the result by the index.
             group_keys: Whether or not to group the keys.
             squeeze: Whether or not to squeeze.
 
         Returns:
             A new DataFrame resulting from the groupby.
         """
+        @ray.remote
+        def assign_partitions(index_df, num_partitions):
```

Comment:
Is there a way to define this remote function (and also `update_columns`) outside of the `DataFrame` class so that we don't define a new remote function every time we call `assign_partitions`? There is some overhead with doing this.